src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensHelpers.cs

```diff
@@ -299,16 +299,21 @@ private static void ConvertMultiLineToSingleLineSpans(SourceText text, Segmented
                 if (classificationType == ClassificationTypeNames.StaticSymbol)
                 {
                     // 4. Token modifiers - each set bit will be looked up in SemanticTokensLegend.tokenModifiers
-                    modifierBits = TokenModifiers.Static;
+                    modifierBits |= TokenModifiers.Static;
                 }
                 else if (classificationType == ClassificationTypeNames.ReassignedVariable)
                 {
                     // 5. Token modifiers - each set bit will be looked up in SemanticTokensLegend.tokenModifiers
-                    modifierBits = TokenModifiers.ReassignedVariable;
+                    modifierBits |= TokenModifiers.ReassignedVariable;
+                }
+                else if (classificationType == ClassificationTypeNames.ObsoleteSymbol)
+                {
+                    // 6. Token modifiers - each set bit will be looked up in SemanticTokensLegend.tokenModifiers
+                    modifierBits |= TokenModifiers.ObsoleteSymbol;
```

Comment:
do we have lsp specific tests to validate the data is at least what we expect?

Reply:
We have TestCompletionForObsoleteSymbol added in this PR, which may cover what you are looking for?